google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -632,27 +611,12 @@ private T performDelete(
 
             List<WriteResult> result = new ArrayList<>();
 
-            Preconditions.checkState(
-                request.getWritesCount() == writeResults.size(),
-                "Expected one write result per operation, but got %s results for %s operations.",
-                writeResults.size(),
-                request.getWritesCount());
-
             Iterator<Mutation> mutationIterator = mutations.iterator();
             Iterator<com.google.firestore.v1.WriteResult> responseIterator =
                 writeResults.iterator();
 
             while (mutationIterator.hasNext()) {
-              Mutation mutation = mutationIterator.next();
-
-              // Don't return both write results for a write that contains a transform, as the fact
-              // that we have to split one write operation into two distinct write requests is an
-              // implementation detail.
-              if (mutation.document != null && mutation.transform != null) {
-                // The document transform is always sent last and produces the latest update time.
-                responseIterator.next();
-              }
-
+              mutationIterator.next();
```

Comment:
Might be simpler now if we just replace it with a `for` loop.

Reply:
You should iterator over `writeResults` and drop `mutationIterator` altogether.

Reply:
After changing it to iterate through `writeResults`, intellij recommended that it be converted to an "enhanced for loop".